src/FrankaLightWeightInterface.cpp

```diff
@@ -44,9 +49,24 @@ void FrankaLightWeightInterface::run_controller() {
   if (this->is_connected()) {
     // restart the controller unless the node is shutdown
     while (!this->is_shutdown()) {
-      std::cout << "Starting controller..." << std::endl;
       try {
-        this->run_joint_torques_controller();
+        switch (this->control_type_) {
+          case proto::JOINT_TORQUE:
+            std::cout << "Starting joint torque controller..." << std::endl;
+            this->run_joint_torques_controller();
+            break;
+          case proto::CARTESIAN_TWIST:
+            std::cout << "Starting cartesian velocities controller..." << std::endl;
+            this->run_cartesian_velocities_controller();
+            break;
+          default:
+            std::cout << "Unimplemented control type! (" << this->control_type_ << ")" << std::endl;
+            [[fallthrough]];
```

Comment:
What does fallthrough do?

Reply:
https://en.cppreference.com/w/cpp/language/attributes/fallthrough

Reply:
so to answer your question, when reaching the `default` condition there is no `break`, so the switch continues to test the `case proto::NONE` (default behavior when there is no `break`). Normally, a compiler issues a warning of that. But here this is actually intentional as I guess @eeberhard wants to start the state publisher even when there is an unrecognized controller. So `fallthrough` disables this warning.

I also understand from this that, first the `proto::NONE` will issue the warning `"Unimplemented control type! (" << this->control_type_ << ")"` which I am not sure is desired, second the joint and cartesian controllers actually break so the publisher will not be called. Is that correct?